src/jaxsim/api/model.py

```diff
@@ -576,6 +576,41 @@ def generalized_free_floating_jacobian(
     return O_J_WL_I
 
 
+@functools.partial(jax.jit, static_argnames=["output_vel_repr"])
+def generalized_free_floating_jacobian_derivative(
+    model: JaxSimModel,
+    data: js.data.JaxSimModelData,
+    *,
+    output_vel_repr: VelRepr | None = None,
+) -> jtp.Matrix:
+    """
+    Compute the free-floating jacobian derivatives of all links.
+
+    Args:
+        model: The model to consider.
+        data: The data of the considered model.
+        output_vel_repr:
+            The output velocity representation of the free-floating jacobian derivatives.
+
+    Returns:
+        The `(nL, 6, 6+dofs)` array containing the stacked free-floating
+        jacobian derivatives of the links. The first axis is the link index.
+    """
+
+    output_vel_repr = (
+        output_vel_repr if output_vel_repr is not None else data.velocity_representation
+    )
+
+    with data.switch_velocity_representation(output_vel_repr):
+        O_J̇_WL_I = jax.vmap(js.link.jacobian_derivative, in_axes=(None, None))(
+            model,
+            data,
+            link_index=jnp.arange(model.number_of_links()),
+        )
```

Comment:
I think that in the way you wrote it, the `I` frame is not correct. Here below my suggestion (you still need to remove the vmapping over `output_vel_repr`):

```suggestion
    O_J̇_WL_I = jax.vmap(js.link.jacobian_derivative, in_axes=(None, None))(
        model,
        data,
        link_index=jnp.arange(model.number_of_links()),
        output_vel_repr=output_vel_repr,
    )
```

Reply:
Arguments passed as kwargs are automatically batched over the first dimension, for this reason I bypassed that argument by switching the velocity representation of data. One idea might be to create a dummy array of `output_vel_repr`, but there must be a smarter way to do this